### ISSUE TITLE
Allow post construction setup phase for Outputer

### ISSUE THREAD

**Issue by Dr15Jones** [hep-cce2/root_serialization#21]
A number of the Outputers need to know what data products will be produced in the job. In the present design, the Outputer has to wait until it sees the first write event call before it can know. It would be better to add another setup step post Outputer construction which passes in the data products information.

This would be useful for the RootEventOutputer as it takes about 60s to setup its Meta data TTree on the first event. This artificially inflates the serial time being recorded.

**Comment by Dr15Jones** [comment]
So there is a post setup phase already `setupForLane`. I moved the creation and filling of the MetaData TTree into that call but it did not decrease the serial time in RootEventOutputer. It seems that there is a costly 'once per job' call happening with the TTree or TBranch and that is dominating the serial time.

@pcanal FYI

**Comment by Dr15Jones** [comment]
So I checked PDS and it has the same ~60s serial time which is constant across all numbers of threads used. So maybe it has something to do with the file system.

**Comment by Dr15Jones** [comment]
As the post setup is possible, I'll close this issue.

**Comment by Dr15Jones** [comment]
I believe I figured out why the serial time is so consistent. When running jobs, I always fill up all the cores. That means when the code is scaling 'per unit time' I am writing to the /scratch system the same amount of MB/s agreggated across all the jobs. So I think I am just seeing the fact that I saturated the storage system at just under 500 MB/s.

I looked at a log file where I ran just 1 single threaded job and processed 24 times as many events as the case where I ran 24 single threaded jobs concurrently. For the 1 job case, the serial time was 1/2 that of the serial time for any of the concurrently running jobs EVEN THOUGH the 1 job ran 24 times as many events.
